src/main.cc

```diff
@@ -178,6 +178,38 @@ signal_cb(ev_loop *loop, struct ev_signal *w, int revents)
 	tarantool_exit(0);
 }
 
+static void
+signal_sigint_cb(ev_loop *loop, struct ev_signal *w, int revents)
+{
+	(void) loop;
+	(void) w;
+	(void) revents;
+
+	/**
+	 * If running in daemon mode, tarantool exits on SIGINT.
+	 * Complain about possibly sudden and unexpected death.
+	 */
+	if (pid_file) {
+		say_crit("got signal %d - %s", w->signum, strsignal(w->signum));
+		tarantool_exit(0);
+	}
```

Comment:
I would return after calling `tarantool_exit(0);`, otherwise we can continue execution of the signal handler: the exit function spawns a fiber, which may yield, and we'll continue execution here (or kinda).

I found it, because I'm going from interactive mode to background and saw `^C` in the log after SIGINT. Not a real problem, I guess, but it would be better to exit cleanly.

Reply:
Fixed.